src/components/chip-widget/chip-widget.tsx

```diff
@@ -16,59 +17,60 @@ const ChipWidget: React.FC<ChipWidgetProps> = ({ columns, onSetFilters }) => {
     // const [groupBy, setGroupBy] = useState<string | null>(null);
     // const [sort, setSort] = useState<{ column: string; order: 'ASC' | 'DESC' } | null>(null);
 
-    const [selectedColumns, setSelectedColumns] = useState<WidgetColumn[]>([]);
+    const [selectedColumns, setSelectedColumns] = useState<WidgetColumnSelected[]>([]);
 
     const columnSelect = (column: WidgetColumn) => {
-        setSelectedColumns([...selectedColumns, column]);
+        setSelectedColumns([
+            ...selectedColumns,
+            {
+                ...column,
+                id: uuid(),
+            },
+        ]);
     };
 
-    const applyFilter = (newFilter: WidgetFilter, index: number) => {
-        setFilters((prevFilters) => {
-            const newFilters = [...prevFilters];
-            newFilters[index] = newFilter; // Update the filter at the given index
-            return newFilters;
+    const applyFilter = (newFilter: WidgetFilter, id: string) => {
+        setSelectedColumns((prevColumns) => {
+            const newColumns = [...prevColumns];
+            const columnIndex = newColumns.findIndex((column) => column.id === id);
+            newColumns[columnIndex].filter = newFilter; // Update the filter at the given index
+            return newColumns;
         });
     };
 
-    useEffect(() => {
-        onSetFilters(filters);
-    }, [filters]);
-
-    const removeColumn = (index: number) => {
+    const removeColumn = (id: string) => {
         setSelectedColumns((prevColumns) => {
             const newColumns = [...prevColumns];
-            newColumns.splice(index, 1); // Remove the column at the given index
+            const columnIndex = newColumns.findIndex((column) => column.id === id);
+            newColumns.splice(columnIndex, 1); // Remove the column at the given index
             return newColumns;
         });
-        setFilters((prevFilters) => {
-            const newFilters = [...prevFilters];
-            newFilters.splice(index, 1); // Remove the filter at the given index
-            return newFilters;
-        });
         onSetFilters(filters);
     };
 
     useEffect(() => {
-        if (selectedColumns.length > 0) {
-            const lastIndex = selectedColumns.length - 1;
-            chipRefs.current[lastIndex]?.focus();
-        }
+        // if (selectedColumns.length > 0) {
```

Comment:
stopped calling focus on the forwardRef. Not necessary any more for control flow, and was causing issues.